sound/soc/soc-core.c

```diff
@@ -1997,7 +2000,7 @@ static void soc_cleanup_card_resources(struct snd_soc_card *card)
 	soc_remove_link_dais(card);
 
 	for_each_card_links_safe(card, link, _link)
-		snd_soc_remove_dai_link(card, link);
+		snd_soc_remove_dai_link(card, link, true);
```

Comment:
@plbossart isnt the simpler thing to just remove soc_unbind_dai_link() from remove_dai_link() and add it after the call to soc_unbind_dai_link()?

Reply:
that's what the code initially did, so that's essentially a revert, yes.